### PR TITLE
looping through dictionaries challenge

### PR DESCRIPTION
**Pull request by rochelleterman**
This is for the Software Carpentry's Instructor Training.

### CODE REVIEW

File: 02-loop.md [hunk 1]
```diff
@@ -228,3 +228,25 @@ so we should always use it when we can.
...
+> Looping through a *string* iterates through character by character.
+> Looping through a *list* iterates through item by item. 
+> We can also loop through python *dictionaries*. 
+> The default looping behavior iterates through each key in the dictionary.
+> 
+> ~~~ {.python}
+> d = {'apples': 0.49, 'oranges': 0.99, 'pears': 1.49, 'bananas': 0.32}
+> #default looping
+> for key in d:
+>    print key, d[key]
```
Comment (QuLogic) [inline 02-loop.md]: `print` is a function in Python 3 (as this lesson teaches).

**Comment by iglpdc** [comment]
I like the challenge, @rochelleterman. What do you think about adding another line at the end to inviting to explore other, more pythonic ways to iterate over the dictionary. You could ask them to check the help for `dict.items()` or `dict.values()`.

**Comment by rochelleterman** [comment]
Hi @iglpdc - thanks for those suggestions. I've gone and edited the challenge to include an extra challenge on `dict.items()` and `dict.values()`, as well as fixing the `print` function, as @QuLogic point out.

File: 02-loop.md [hunk 2]
```diff
@@ -228,3 +228,29 @@ so we should always use it when we can.
...
+> d = {'apples': 0.49, 'oranges': 0.99, 'pears': 1.49, 'bananas': 0.32}
+> #default looping
+> for key in d:
+>    print(key, d[key])
+> ~~~
+> 
+> Knowing this, use a for loop to calculate how much it'll cost you 
+> to buy 2 pieces of each fruit.
+>
+> Extra Challenge: Python allows others ways to iterature over a dictionary.
```
Comment (QuLogic) [inline 02-loop.md]: iterature -> iterate

**Comment by gvwilson** [comment]
We definitely need to explain how to loop over dictionaries - but at the moment, this lesson doesn't introduce dictionaries at all. We *are* going to introduce them in https://github.com/swcarpentry/python-novice-gapminder - could you please re-submit this PR there?

**Pull request closed without merging**